python/cuml/dask/preprocessing/label.py

```diff
@@ -0,0 +1,284 @@
+    category_num : int
+        Number of categories of label encoder
+    '''
+    ord_label = y.unique()
+    ord_max = ord_label.max()
+    ord_min = ord_label.min()
+    if ord_min < 0:
+        raise ValueError(
+            'y contains ordianl unseen label {}'.format(ord_min))
+    if ord_max >= category_num:
+        raise ValueError(
+            'y contains previously unseen label {}'.format(ord_max))
+
+
+def _ordinal_to_str(y, label_encoder, category_num):
+    ''' Help convert ordinal labels to string labels
+
+    Parameters
+    ----------
+    y : cudf.Series
+        The series to be inverse_transformed
+    label_encoder: LabelEncoder object
+        Fitted LabelEncoder object
+    category_num : int
+        Number of categories of label encoder
+    '''
+    # check if y's dtype is np.int32, otherwise convert it
+    y = _check_npint32(y)
+
+    # check if ord_label out of bound
+    _check_ordinal_label(y, category_num)
+
+    # convert ordinal label to string label
+    return cudf.Series(label_encoder._cats.gather_strings(
+        y.data.mem.device_ctypes_pointer.value, len(y)))
+
+
+class LabelEncoder(object):
+    ''' Encode labels with value between 0 and n_classes-1
+
+    Examples
+    --------
+    >>> from dask_cuml.preprocessing import LabelEncoder
+    >>> import dask_cudf
+    >>> import cudf
+
+    >>> data = cudf.Series(['a', 'b', 'a', 'b', 'c', 'd', 'a'])
+    >>> data = dask_cudf.from_cudf(data, npartitions=2)
+
+    >>> le = LabelEncoder()
+    >>> le.fit(data)
+    >>> print(le._cats.keys())
+    ['a', 'b', 'c', 'd']
+
+    >>> encoded = le.transform(data)
+    >>> print(encoded.compute())
+    0    0
+    1    1
+    2    0
+    3    1
+    0    2
+    1    3
+    2    0
+    dtype: int32
+
+    >>> ord_label = cudf.Series([0, 0, 1, 2, 1])
+    >>> ord_label = dask_cudf.from_cudf(data, npartitions=2)
+    >>> str_label = le.inverse_transform(ord_label)
+    >>> print(type(str_label))
+    <class 'cudf.dataframe.series.Series'>
+
+    >>> print(str_label)
+    0    a
+    1    a
+    2    b
+    3    c
+    4    b
+    dtype: object
+    '''
+
+    def __init__(self, *args, **kwargs):
+        self._cats = None
+        self._dtype = None
+        self._fitted = False
+
+    def _check_is_fitted(self):
+        ''' Check whether the LabelEncoder object has been fitted and
+        is ready to transform, raise a ValueError otherwise.
+        '''
+        if (not self._fitted) or (self._cats is None):
+            raise ValueError('LabelEncoder must be fit first')
+
+    def fit(self, y):
+        ''' Fit label encoder
+
+        Parameters
+        ----------
+        y : dask_cudf.Series or cudf.Series
+            Target values
+
+        Returns
+        -------
+        self : returns an instance of self
+        '''
+        if isinstance(y, dask_cudf.Series):
+            y = y.map_partitions(_enforce_str)
+            self._cats = nvcategory.from_strings(y.compute().data)
```

Comment:
If I'm understanding this correctly, this is assuming that the set of labels in the dask Series is small enough to fit on a single GPU. What happens when this is not the case?

Reply:
Thanks for the review, it makes sense.
Unfortunately, my internship just ended last week and I lost access to DGX box. Maybe this PR could be assigned to another member and modifications can be made on the current basis